R/sim_contacts.R

```diff
@@ -0,0 +1,82 @@
+#' Simulate contacts for an infectious disease outbreak
+#'
+#' @inheritParams sim_linelist
+#' @param contact_tracing_status_probs A named `numeric` vector with the
+#' probability of each contact tracing status. The names of the vector must
+#' be `"under_followup"`, `"lost_to_followup"`, `"unknown"`. Values of each
+#' contact tracing status must sum to one.
+#'
+#' @return A contacts `<data.frame>`
+#' @export
+#'
+#' @author Joshua W. Lambert, Carmen Tamayo
+#'
+#' @examples
+#' # load data required to simulate contacts
+#' contact_distribution <- epiparameter::epidist(
+#'   disease = "COVID-19",
+#'   epi_dist = "contact distribution",
+#'   prob_distribution = "pois",
+#'   prob_distribution_params = c(mean = 2)
+#' )
+#'
+#' contact_interval <- epiparameter::epidist(
+#'   disease = "COVID-19",
+#'   epi_dist = "contact interval",
+#'   prob_distribution = "gamma",
+#'   prob_distribution_params = c(shape = 1, scale = 1)
+#' )
+#'
+#' contacts <- sim_contacts(
+#'   contact_distribution = contact_distribution,
+#'   contact_interval = contact_interval,
+#'   prob_infect = 0.5
+#' )
+sim_contacts <- function(contact_distribution,
+                         contact_interval,
+                         prob_infect,
+                         outbreak_start_date = as.Date("2023-01-01"),
```

Comment:
Would it be useful for the default date to be the current date?

Reply:
This would cause some knock-on effects with updates to the `README` and other documentation that use `sim_*()` function when they are rendered as they would be updated each time the render was triggered. I don't think the benefit of having the current date outweighs this downside. 

However, I'm happy to add an example to the Getting started vignette on supplying `Sys.Date()` to `outbreak_start_date` to make users aware of how starting an outbreak from that given day can easily be achieved. Let me know what you think to this.

Reply:
I think it's not a huge issue, so it's probably fine to keep this as it is.